src_c/rect.c

```diff
@@ -956,25 +956,42 @@ pg_rect_clipline(pgRectObject *self, PyObject *args)
     return Py_BuildValue("((ii)(ii))", x1, y1, x2, y2);
 }
 
-static PyObject *
-pg_rect_contains(pgRectObject *self, PyObject *args)
-{
-    int contained;
-    GAME_Rect *argrect, temp;
-
-    if (!(argrect = pgRect_FromObject(args, &temp))) {
-        return RAISE(PyExc_TypeError, "Argument must be rect style object");
+static int
+_pg_rect_contains(pgRectObject *self, PyObject *arg)
+{
+    GAME_Rect *argrect, temp_arg;
+    if (!(argrect = pgRect_FromObject((PyObject *)arg, &temp_arg))) {
+        // Case where argument is one of (x,y,w,h)
+        if (PyLong_Check(arg)) {
```

Comment:
I don't think this part of the code should run when `pg_rect_contains` calls this function because `Rect.contains` is not supposed to handle integer coordinates.

Reply:
Alright, I think then the best way is to make a base helper function which both the explicit and implicit contains methods call to make sure the behavior is slightly different for each. Pushed a commit like that, let me know what you think!

Reply:
Yes, good idea. And nice fix!